### ISSUE TITLE
Motivational SamNonDuePin example does not work against M2 configuration

### ISSUE THREAD

**Issue by adamvoss** [macchina/arduino-boards-sam#20]
Using the first [motivation example from the README](https://github.com/macchina/arduino-boards-sam#motivation) the code behaves as expected when uploading using the "Arduino Due (Native USB Port)" configuration.  However, when I upload using "Macchina M2 (Beta)", only the Yellow LED will turn on.

I used `SerialUSB.println` and it appears the likely culprits `X0`, `X1`, and `PIN_EMAC_ERX1` are the same either way so likely not the cause.

**Comment by TDoust** [comment]
Hi Adam

All examples referring to the SamnonDue pin references should be removed from the website because the SamnonDue was a quick hack done by someone prior to me to get the M2 non-standard pins working with a standard DUE variant file & the SamnonDue variant file. What you are trying to do is use the modified M2variant file with the SamnonDue file & this will cause major confusion & iam surprised that was your only issue.

Now that the Variant files have been made specifically for the Macchina M2 the SamnonDue reference is no longer needed & therefore should be removed from the website along with any examples using the SamnonDue otherwise this may cause confusion to newbies in which to use SamnonDue or the new variant files.

The only difference between the release M2 & the beta is the 6 source & 6 sink output pins. The new M2 specific variant files takes care of the Beta & release version of the M2 by using the define Beta for those that own a Beta release & wish to use the source & sink pins together with the risk of destroying the driver board (as I have done twice during testing of the 12vIO driver library) if not using the M2_12VIO driver library I have developed & still testing & improving.

**Comment by TDoust** [comment]
Hi Adam

I have just compiled the example 2 sketch using Arduino 1.8.3 IDE with the following errors

```
C:\Program Files (x86)\Arduino\arduino-builder -dump-prefs -logger=machine -hardware C:\Program Files (x86)\Arduino\hardware -hardware C:\Users\Tony\AppData\Local\Arduino15\packages -hardware C:\Users\Tony\Documents\Arduino\hardware -tools C:\Program Files (x86)\Arduino\tools-builder -tools C:\Program Files (x86)\Arduino\hardware\tools\avr -tools C:\Users\Tony\AppData\Local\Arduino15\packages -built-in-libraries C:\Program Files (x86)\Arduino\libraries -libraries C:\Users\Tony\Documents\Arduino\libraries -fqbn=macchina:sam:m2 -ide-version=10803 -build-path C:\Users\Tony\AppData\Local\Temp\arduino_build_671103 -warnings=none -build-cache C:\Users\Tony\AppData\Local\Temp\arduino_cache_709043 -prefs=build.warn_data_percentage=75 -prefs=runtime.tools.arm-none-eabi-gcc.path=C:\Users\Tony\AppData\Local\Arduino15\packages\macchina\tools\arm-none-eabi-gcc\4.8.3-2014q1 -prefs=runtime.tools.bossac.path=C:\Users\Tony\AppData\Local\Arduino15\packages\macchina\tools\bossac\1.6.1-arduino -verbose C:\Users\Tony\Documents\Arduino\Example\Example\Example.ino 

Board m2 (platform sam, package macchina) is unknown

Error compiling for board Macchina M2.
```
 

However the program compiles & uploads Ok when using Visual studio however #include <Arduino.h> does not find the file or any of the standard include files i.e

```
#include <Arduino.h>
#include <pins_arduino.h>
#include <stdint.h>
```

I think there may be something missing in the boards config files breaking the compile using the Arduino IDE

**Comment by adamvoss** [comment]
> I have just compiled the example 2 sketch using Arduino 1.8.3 IDE with the following errors [...]

I assume by `example 2` you mean the second motivational example.  I am using a newer version of the Arduino IDE but it compiled (M2 and M2-B), uploaded and runs (M2-B) just fine for me.  The error you gave and the described behavior makes me suspect that the problem exists in the setup/configuration of your Arduino IDE rather than the board configuration itself.  I suspect you will have problems with any sketch if you try using that Macchina M2 configuration.

My recommendation would be to delete any custom board configurations (both those added via the Board Manager and any you manually setup for development), install again from the Board Manager and see if it works.

**Comment by adamvoss** [comment]
> All examples referring to the SamnonDue pin references should be removed from the website because the SamnonDue was a quick hack done by someone prior to me to get the M2 non-standard pins working with a standard DUE variant file & the SamnonDue variant file.

I am not advocating that `SamNonDuePin` should be the path forward for development, but am trying to utilize existing code to test the custom board definition.  `SamNonDuePin` is a proven library which predates the M2 and existing M2 code and projects still use it.

This sounds like the same issue as https://github.com/macchina/arduino-boards-sam/pull/18#issuecomment-325079964, https://github.com/macchina/Arduino_Due_SD_HSMCI/pull/3#issuecomment-325080040, and [Should Due sketches run on the M2 where possible?](https://forum.macchina.cc/t/should-due-sketches-run-on-the-m2-where-possible/252?u=adamvoss).

If a reason has been identified to break compatibility, then we document the reasoning and implications and have an easy answer for why things do not work.  If this is not known, then it is not easily to say if something is broken in the custom board configuration or in the supplied sketch.

**Comment by adamvoss** [comment]
I have traced it so far to see that the reading of the pins and conditional logic is all working right.  However the call `digitalWrite(Red, LOW)` is not having the desired effect.

**Comment by TDoust** [comment]
Adam

If you are still trying to get the first example working

You need to select a standard DUE native USB port in the boards & then compile using samnondue against this standard due

It will work then

**Comment by TDoust** [comment]
Hi Adam

Found my problem with the Arduino IDE

I had to remove Arduino IDE totally as well as the C:\Users\Tony\AppData\Local\Arduino15 etc directory only leaving the Arduino directory off the documents directory & then reinstall Arduino IDE

In other words an almost clean install

This was an interesting exercise as after installing Arduino IDE & the Due boards definitions then restarting & installing Macchina Definition Then restarting

After the restart I plugged in the MACCHINA M2 & the IDE would not install the correct windows serial drivers to recognise the M2 correctly

This was a worthwhile exercise & I recommend you try this as our documentation does not have any reference to this problem & how to get around this by installing the correct serial drivers manually

I would imagine a clean install may have the same problems as I remember reading somewhere this was an issue with the DUE & there was a fix by installing the drivers manually. Can’t remember where I had read this

Regards Tony Doust

**Comment by adamvoss** [comment]
I'm glad to hear you got it working!  [This the link into our documentation covering drivers](http://docs.macchina.cc/m2/getting-started/arduino.html#drivers).  Do you have suggested improvements?

**Comment by TDoust** [comment]
Adam

That’s all good re the drivers providing they have downloaded the zip file of the IDE & not the windows installer version

Regards Tony Doust

**Comment by adamvoss** [comment]
Resolved by #26
